ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java

```diff
@@ -416,6 +416,12 @@ public void close() {
       LOG.warn(getId() + ": Failed to shutdown implExecutor", ignored);
     }
 
+    try {
+      ConcurrentUtils.shutdownAndWait(executor);
+    } catch (Exception ignored) {
+      LOG.warn(getId() + ": Failed to shutdown executor", ignored);
+    }
+
```

Comment:
We should move `implExecutor` inside `checkStateAndClose(..)` instead.  And also initialize `implExecutor` and `executor` in the `startImpl()` method.

Reply:
Great Idea! Has fixed